src/System.Net.NetworkInformation/tests/FunctionalTests/NetworkInterfaceBasicTest.cs

```diff
@@ -37,7 +37,16 @@ public void BasicTest_AccessInstanceProperties_NoExceptions()
                 _log.WriteLine("Type: " + nic.NetworkInterfaceType);
                 _log.WriteLine("Status: " + nic.OperationalStatus);
                 _log.WriteLine("Speed: " + nic.Speed);
-                Assert.True(nic.Speed >= 0, "Overflow");
+
+                if (nic.OperationalStatus != OperationalStatus.Down)
+                {
+                    Assert.True(nic.Speed >= 0, "Overflow");
+                }
+                else
+                {
+                    Assert.True(nic.Speed >= 0 || nic.Speed == -1, "Overflow");
```

Comment:
And then as with the above suggestion, it could instead be:
```C#
Assert.InRange(nic.Speed, -1, long.MaxValue);
```
And if you really wanted to, you could then combine both of these back together, e.g.
```C#
Assert.InRange(nic.Speed, nic.OperationalStatus != OperationalStatus.Down ? 0 : -1, long.MaxValue);
```
Just a suggestion, up to you.